### ISSUE TITLE
COCO OKS Metrics Usage

### ISSUE THREAD

**Issue by Naman-ntc** [Microsoft/human-pose-estimation.pytorch#27]
Hi, I am unable to understand how OKS is calculated in experiments using COCO dataset.
In the train function in `lib/core/function.py` you seem to call `accuracy` from the file `lib/core/evaluate.py`. But that accuracy is PCKh right? So how do you calculate OKS.

Could you please explain the steps how can I calculate OKS given I use your dataloader?? Thanks alot in advance!!!

**Comment by leoxiaobin** [comment]
We use a simple PCKh metric to evaluate our training procedure, and we only use OKS for validation procedure, you can look into the code at https://github.com/Microsoft/human-pose-estimation.pytorch/blob/d69ed56bdbc1f16a288921e302c87fcb33554e37/lib/dataset/coco.py#L273.

**Comment by Naman-ntc** [comment]
Does this piece of code perform the validation:
https://github.com/Microsoft/human-pose-estimation.pytorch/blob/d69ed56bdbc1f16a288921e302c87fcb33554e37/lib/core/function.py#L180-L182

**Comment by Naman-ntc** [comment]
Yeah i finally got my hands dirty with it. I was able to implement and train stacked hourglass with it. Thanks for putting the wonderful code and such great response

**Comment by hengck23** [comment]
it seems that for pckh, you are using reference_size =[0.1*H, 0.1*W] in your code.

distance_x/(0.1*H)
distance_y/(0.1*W)

the reference_size for x,y seems to be flipped by mistake?


---

see evaluate.py: 

 def accuracy(output, target, hm_type='gaussian', thr=0.5):

... norm = np.ones((pred.shape[0], 2)) * np.array([h, w]) / 10

**Comment by YuQi9797** [comment]
Is there a handwritten version of OKS here? Instead of calling the API .
